cohorts/load.py

```diff
@@ -397,14 +412,17 @@ def is_lambda(func):
             for key, value in on.iteritems():
                 col, df = apply_func(on=value, col=key, df=df)
                 cols.append(col)
-            return (cols, df)
         if type(on) == list:
             cols = []
             for i, elem in enumerate(on):
                 col = elem.__name__ if not is_lambda(elem) else "column_%d" % i
                 col, df = apply_func(on=elem, col=col, df=df)
                 cols.append(col)
-            return (cols, df)
+
+        if (rename_cols):
+            df = df.rename(columns=_strip_column_names(d.columns,
+                   keep_paren_contents=keep_paren_contents))
```

Comment:
@tavinathanson wasn't sure here what I should do with `cols` - welcome your advice on this part